_posts/2021-01-21-ooops-can-I-try-again-please.md

```diff
@@ -0,0 +1,73 @@
+---
+layout: post
+header_image: /img/posts/ooops-can-I-try-again-please/header.jpg
+title: "Ooops, can I try again, please?"
+author: Mauro Servienti
+synopsis: "When systems fail, we can retry the whole process and be successful. However, there are scenarios in which retrying a subset of the process might be a better choice. Not all failures are born equal."
+tags:
+- Distributed Systems
+- Retries
+- Errors
+- Failures
+---
+
+It failed. Systems fail all the time: Bugs, unavailability of required resources, hardware failures, and many more reasons cause systems to fail. When the system uses messages and queues to exchange information and drive behaviors, one of the advantages is that we can retry messages. A message comes in, and the message handling process fails. The message can be returned to the queue and tried again. If message processing logic changes some data, it can rollback the corresponding database transaction. So far, so good, in many scenarios, we can retry messages.
```

Comment:
```suggestion
Systems fail all the time: Bugs, unavailability of required resources, hardware failures, and many more reasons cause systems to fail. When the system uses messages and queues to exchange information and drive behaviors, one of the advantages is that we can retry messages. A message comes in, and the message handling process fails. The message can be returned to the queue and tried again. If message processing logic changes some data, it can rollback the corresponding database transaction. So far, so good, in many scenarios, we can retry messages.
```

> A message comes in, and the message handling process fails. The message can be returned to the queue and tried again.

Might be worth explaining that it can be tried again _after_ the failure has been addressed.